mne/encoding/feature.py

```diff
@@ -0,0 +1,341 @@
+        pre-defined delays. Negative means timepoints in the past,
+        positive means timepoints in the future.
+    sfreq: float
+        The sampling frequency of the series. Defaults to 1.
+    """
+    def __init__(self, delays=None, time_window=None, sfreq=1.):
+        # Check if we need to creafte delays ourselves
+        if time_window is not None:
+            if sfreq is None:
+                raise ValueError('If time_window given, must give sfreq.')
+            if delays is not None:
+                raise ValueError('Delays must be None if time_window is given')
+            tmin, tmax = time_window
+            start_idx = int(round(tmin * sfreq))
+            delays = np.arange(start_idx, int(round(tmax * sfreq)) + 1) / sfreq
+        elif delays is None:
+            raise ValueError('One of `time_window` or `delays` must be given.')
+        self.delays = np.asarray(delays)
+        self.sfreq = float(sfreq)
+        self.feature_type = 'delay'
+
+    def fit(self, X, y=None):
+        """Create a time-lagged representation of X.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_times, n_features)
+            The input data to be time-lagged and returned.
+        """
+        X_del, names = _delay_timeseries(X, self.delays, sfreq=self.sfreq)
+        self.X_delayed_ = X_del
+        self.names_ = names
+
+    def transform(self, X=None, y=None):
+        """See `fit` method."""
+        return self.X_delayed_
+
+    def fit_transform(self, X, y=None):
+        """See `fit` method."""
+        self.fit(X)
+        return self.transform()
+
+
+class EventsBinarizer(object):
+    """Create a continuous-representation of event onsets.
+
+    Parameters
+    ----------
+    n_times : int
+        The total number of times in the output array.
+    sfreq : float
+        The sampling frequency used to convert times to sample indices.
+        Defaults to 1.
+    sparse : bool
+        Whether to output continuous events as a dense array or a sparse
+        matrix. Defaults to False.
+    """
+    def __init__(self, n_times, sfreq=1., sparse=False):
+        self.n_times = n_times
+        self.sfreq = sfreq
+        self.feature_type = 'event_id'
+        self.sparse = sparse
+
+    def fit(self, event_ixs, event_ids=None, event_dict=None):
+        """Binarize the events.
+
+        Parameters
+        ----------
+        event_ixs : array, shape (n_events)
+            The event indices or onsets in seconds. `If self.sfreq == 1`,
+            then these are assumed to be indices. If `self.sfreq != 1`, these
+            should be onsets in seconds and will be multiplied by `sfreq`.
+        event_ids : array-like, shape (n_events) | None
+            The event ID for each event. If None, all events are assumed to be
+            of same type.
+        event_dict : dict
+            A dictionary of (event_id: event_id_name) pairs. Defines a string
+            name for each event type.
```

Comment:
I imagined this module being able to do what Jona suggests here, but using the pipelines instead of functions. We could make a general high-level function like `make_predictors`, but I've been trying to keep things based off of class instances thus far, as opposed to a single high-level function like `make_predictors`. But I'm open to suggestions so we can discuss next week...